django/contrib/postgres/indexes.py

```diff
@@ -1,8 +1,38 @@
 from __future__ import unicode_literals
 
-from django.db.models import Index
+from django.db.models.indexes import Index
 
-__all__ = ['GinIndex']
+__all__ = ['BrinIndex', 'GinIndex']
+
+
+class BrinIndex(Index):
+
+    suffix = 'brin'
+
+    def __init__(self, fields=[], name=None, pages_per_range=128):
+        self.pages_per_range = pages_per_range
+        return super(BrinIndex, self).__init__(fields, name)
+
+    def __repr__(self):
+        return "<%(name)s: fields=%(fields)s, pages_per_range=%(pages_per_range)s>" % {
+            'name': self.__class__.__name__,
+            'fields': "'{}'".format(', '.join(self.fields)),
+            'pages_per_range': self.pages_per_range,
+        }
+
+    def deconstruct(self):
+        path, args, kwargs = super(BrinIndex, self).deconstruct()
+        kwargs['pages_per_range'] = self.pages_per_range
+        return path, args, kwargs
+
+    def create_sql(self, model, schema_editor):
+        using = ' USING brin '
+        specials = 'WITH (pages_per_range = %d) ' % self.pages_per_range
+        parameters = self.index_parameters(model, schema_editor, using)
+        parameters['specials'] = specials
+        sql_create_index = "CREATE INDEX %(name)s ON %(table)s%(using)s (%(columns)s)%(specials)s%(extra)s"
```

Comment:
There are quite a few places that would be affected by this. An alternative way to deal with this could be to introduce one more ``sql_create_index_sql_parameterized`` that has this ``with``-parameter. It would mean fewer ``try/except``-blocks etc. to handle the replacement.